src/ssh/cockpitsshrelay.c

```diff
@@ -556,11 +556,11 @@ session_has_known_host_in_file (const gchar *file,
                                 const guint port)
 {
 #if !HAVE_DECL_SSH_SESSION_HAS_KNOWN_HOSTS_ENTRY
-  shim_set_knownhosts_file(file);
+  shim_set_knownhosts_file (file ? file : data->user_known_hosts);
 #endif
   /* HACK: https://bugs.libssh.org/T108 */
   if (!file)
-      g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_SSH_DIR, NULL) == SSH_OK);
+    g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_SSH_DIR, NULL) == SSH_OK);
   g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_KNOWNHOSTS, file) == SSH_OK);
```

Comment:
BTW, shouldn't that be `SSH_OPTIONS_GLOBAL_KNOWNHOSTS`?

Reply:
Oh yea you're right, after T107 we can use that option.